LCAPI.Core/ModData.cs

```diff
@@ -0,0 +1,186 @@
+﻿namespace LCAPI.Core;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using static IngamePlayerSettings;
+
+/// <summary>
+/// Class that handle save/loading custom data from mod.
+/// </summary>
+public sealed class ModData
+{
+    internal static ES3Settings defaultSetting;
+    internal static HashSet<string> tracker = new();
+    internal static void Setup()
+    {
+        defaultSetting = new ES3Settings
+        {
+            location = ES3.Location.File,
+            prettyPrint = true,
+            saveChildren = true,
+            compressionType = ES3.CompressionType.Gzip,
+            bufferSize = 4096,
+            encoding = Encoding.UTF8
+        };
+    }
+
+    internal void SetupConfig()
+    {
+        if (useCustomPath)
+        {
+            config = new($"{customPath}.sav", defaultSetting);
+            return;
+        }
+
+        config = new($"{guid}_LCModData.sav", defaultSetting);
+    }
+
+    internal string guid;
```

Comment:
Unless I'm mistaken, it's good (CSharp) practice to keep all member variables grouped together at the top of a class definition.

E.g.

Class:
- member vars
- Constructor
- member methods

Reply:
this should be fine in my latest commit..